rpc/src/rpc.rs

```diff
@@ -1649,21 +1653,20 @@ impl JsonRpcRequestProcessor {
         signatures: Vec<Signature>,
         config: Option<RpcSignatureStatusConfig>,
     ) -> Result<RpcResponse<Vec<Option<TransactionStatus>>>> {
-        let mut statuses: Vec<Option<TransactionStatus>> = vec![];
-
         let search_transaction_history = config
             .map(|x| x.search_transaction_history)
             .unwrap_or(false);
-        let bank = self.bank(Some(CommitmentConfig::processed()));
-
-        if search_transaction_history && !self.config.enable_rpc_transaction_history {
-            return Err(RpcCustomError::TransactionHistoryNotAvailable.into());
+        if search_transaction_history {
+            self.check_if_transaction_history_enabled()?;
         }
 
+        let bank = self.bank(Some(CommitmentConfig::processed()));
+        let mut statuses: Vec<Option<TransactionStatus>> = vec![];
+
         for signature in signatures {
             let status = if let Some(status) = self.get_transaction_status(signature, &bank) {
                 Some(status)
-            } else if self.config.enable_rpc_transaction_history && search_transaction_history {
+            } else if search_transaction_history {
```

Comment:
We previously confirmed that:
`A) search_transaction_history IMPLIES enable_tx_history`
Here, we search only if:
`B) search_transaction_history AND enable_tx_history`

Due to `A)`, we can reduce `B)` to just `search_transaction_history`